app/controllers/registrations_controller.rb

```diff
@@ -1,6 +1,6 @@
 class RegistrationsController < Devise::RegistrationsController
-  prepend_before_filter :require_no_authentication, only: [:new, :create, :cancel]
-  prepend_before_filter :authenticate_scope!, only: [:edit, :update, :destroy]
+  prepend_before_action :require_no_authentication, only: [:new, :create, :cancel]
+  prepend_before_action :authenticate_scope!, only: [:edit, :update, :destroy]
```

Comment:
I didn't know what the difference between a `before_action` and a `prepend_before_action` was, [so I looked it up](https://stackoverflow.com/a/49791090/5009528).

A `prepend_before_action` runs before a `before_action`. There are no `before_action`s in this controller, so it seems like these prepends could just be regular `before_action`s. 🤔 

Since we're dealing with Devise stuff, and I am not sure of the order of loading or any including that may be happening between controllers, I feel hesitant to make that change. Of course, if the Devise docs say this is the way, then this is the way. I'll do a deeper dive on that in a bit. I've got 80 more files to investigate...

Reply:
Yeah, I looked and that's the default Devise code: https://github.com/heartcombo/devise/blob/main/app/controllers/devise/registrations_controller.rb